### PR TITLE
Apply changes from clang-tidy: modernize-use-default. (Part 3)

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This uses clang-tidy's modernize-use-default tool to replace default bodies of special member functions with `= default;`. I then did quite a bit of cleanup, such as remove default constructors and destructors that would be automatically generated anyway.

**To test:**

code review: If there are missing symbols, the build should fail. I'm not sure how great our test coverage of constructors is. There are a number of classes with private virtual destructors. Double-check I didn't delete any of those, potentially making the destructor public. 

There is no GitHub issue associated with this PR.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Crystal/src/OptimizeCrystalPlacement.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ class OrEnabledWhenProperties : public Kernel::IPropertySettings {
 };
 
 OptimizeCrystalPlacement::OptimizeCrystalPlacement() : Algorithm() {}
```
Comment (SimonHeybrock) [inline Framework/Crystal/src/OptimizeCrystalPlacement.cpp]: Remove?

File: Framework/Crystal/src/ShowPeakHKLOffsets.cpp [hunk 2]
```diff
@@ -26,7 +26,7 @@ namespace Crystal {
 DECLARE_ALGORITHM(ShowPeakHKLOffsets)
 
 ShowPeakHKLOffsets::ShowPeakHKLOffsets() : Algorithm() {}
```
Comment (SimonHeybrock) [inline Framework/Crystal/src/ShowPeakHKLOffsets.cpp]: Remove?

File: Framework/DataHandling/inc/MantidDataHandling/AsciiPointBase.h [hunk 3]
```diff
@@ -43,8 +43,6 @@ class DLLExport AsciiPointBase : public API::Algorithm {
 public:
   /// Default constructor
   AsciiPointBase() : m_qres(0), m_xlength(0), m_ws() {}
```
Comment (SimonHeybrock) [inline Framework/DataHandling/inc/MantidDataHandling/AsciiPointBase.h]: Member init instead?

File: Framework/DataHandling/inc/MantidDataHandling/LoadILL.h [hunk 4]
```diff
@@ -38,7 +38,6 @@ class DLLExport LoadILL : public API::IFileLoader<Kernel::NexusDescriptor> {
 public:
   /// Constructor
   LoadILL(); /// Virtual destructor
```
Comment (SimonHeybrock) [inline Framework/DataHandling/inc/MantidDataHandling/LoadILL.h]: Comment.

File: Framework/DataHandling/inc/MantidDataHandling/LoadRKH.h [hunk 5]
```diff
@@ -49,8 +49,6 @@ class DLLExport LoadRKH : public API::IFileLoader<Kernel::FileDescriptor> {
 public:
   /// Constructor
   LoadRKH() : m_unitKeys(), m_RKHKeys() {}
```
Comment (SimonHeybrock) [inline Framework/DataHandling/inc/MantidDataHandling/LoadRKH.h]: Remove?

File: Framework/DataHandling/inc/MantidDataHandling/LoadQKK.h [hunk 6]
```diff
@@ -41,8 +41,6 @@ class DLLExport LoadQKK : public API::IFileLoader<Kernel::NexusDescriptor> {
 public:
   /// (Empty) Constructor
   LoadQKK() {}
```
Comment (SimonHeybrock) [inline Framework/DataHandling/inc/MantidDataHandling/LoadQKK.h]: Remove?

File: Framework/DataHandling/inc/MantidDataHandling/LoadSassena.h [hunk 7]
```diff
@@ -56,8 +56,6 @@ class DLLExport LoadSassena : public API::IFileLoader<Kernel::NexusDescriptor> {
 public:
   /// Constructor
   LoadSassena() : API::IFileLoader<Kernel::NexusDescriptor>(), m_filename(""){};
```
Comment (SimonHeybrock) [inline Framework/DataHandling/inc/MantidDataHandling/LoadSassena.h]: Remove?

**Comment by SimonHeybrock** [comment]
@quantumsteve Looks good, apart from comments above and many places where you have defaulted destructors, that could probably be removed?

Also note that there are merge conflicts.

File: Framework/CurveFitting/inc/MantidCurveFitting/IFittingAlgorithm.h [hunk 8]
```diff
@@ -50,7 +50,6 @@ namespace CurveFitting {
 class DLLExport IFittingAlgorithm : public API::Algorithm {
 public:
   IFittingAlgorithm();
```
Comment (SimonHeybrock) [inline Framework/CurveFitting/inc/MantidCurveFitting/IFittingAlgorithm.h]: Remove.

**Comment by SimonHeybrock** [comment]
Changes up until here look good.

**Comment by SimonHeybrock** [comment]
@quantumsteve I think our coding standard need an update: "The compiler-generated copy constructor and assignment operator should not be relied upon." (http://www.mantidproject.org/C%2B%2B_Coding_Standards).

File: Framework/DataHandling/src/LoadVulcanCalFile.cpp [hunk 9]
```diff
@@ -39,14 +39,9 @@ const size_t NUMBERRESERVEDPERMODULE = 1250;
...
-      m_badPixFilename(""), m_tofOffsetsWS(), m_offsetsWS(), m_groupWS(),
-      m_maskWS(), m_doAlignEventWS(false), m_eventWS(), m_effLTheta() {}
-
-//----------------------------------------------------------------------------------------------
-/** Destructor
- */
-LoadVulcanCalFile::~LoadVulcanCalFile() = default;
+    : m_instrument(), m_groupingType(VULCAN_OFFSET_BANK), m_tofOffsetsWS(),
+      m_offsetsWS(), m_groupWS(), m_maskWS(), m_doAlignEventWS(false),
+      m_eventWS(), m_effLTheta() {}
```
Comment (SimonHeybrock) [inline Framework/DataHandling/src/LoadVulcanCalFile.cpp]: Remove default constructors calls as you did in other places?

**Comment by SimonHeybrock** [comment]
@quantumsteve Looks good, apart form broken Windows build.

**Comment by SimonHeybrock** [comment]
:shipit:

**Comment by FedeMPouzols** [comment]
Nice cleanup! I'm guessing that our coding standards may benefit from a bit of updating, and emphasizing C++11/14 features?

**Pull request merged**